public/components/custom_panels/helpers/utils.tsx

```diff
@@ -384,6 +391,183 @@ export const renderCatalogVisualization = async ({
   setIsLoading(false);
 };
 
+const createOtelVisualizationMetaData = (
+  documentName: string,
+  visualizationType: string,
+  startTime: string,
+  endTime: string,
+  queryData: object
+) => {
+  return {
+    name: documentName,
+    description: '',
+    query: '',
+    type: visualizationType,
+    subType: PPL_METRIC_SUBTYPE,
+    metricType: OTEL_METRIC_SUBTYPE,
+    selected_date_range: {
+      start: startTime,
+      end: endTime,
+      text: '',
+    },
+    selected_timestamp: {
+      name: 'timestamp',
+      type: 'timestamp',
+    },
+    selected_fields: {
+      text: '',
+      tokens: [],
+    },
+    userConfigs: {
+      layout: dynamicLayoutFromQueryData(queryData),
+    },
+  };
+};
+
+export const fetchAggregatedBinCount = (
+  minimumBound: string,
+  maximumBound: string,
+  startTime: string,
+  endTime: string,
+  documentName: string,
+  http: CoreStart['http'],
```

Comment:
Please remove passing http client as param here. You can use `getOSDhttp` from anywhere in the code. Example usage: https://github.com/opensearch-project/dashboards-observability/blob/177ae3142f467cd1763cc4f876a311085ea3fd80/public/components/notebooks/components/paragraph_components/para_output.tsx#L128

Reply:
changed it